source/agora/test/Base.d

```diff
@@ -1028,3 +1034,70 @@ public bool containSameBlocks (APIS)(APIS nodes, size_t height)
 
     return true;
 }
+
+
+/*******************************************************************************
+
+    Generate a genesis block.
+
+    For each key pair, a freeze transaction and an Enrollment
+    will be created and added to the generated Genesis block.
+    The first transaction is the `UnitTestGenesisTransaction`
+
+    Params:
+        key_pairs = key pairs for signing enrollments with
+
+    Returns:
+        The genesis block
+
+*******************************************************************************/
+
+private immutable(Block) makeGenesisBlock (in KeyPair[] key_pairs)
+{
+    // 1 payment tx, the rest are freeze txs
+    Transaction[] txs;
+    txs ~= cast(Transaction)UnitTestGenesisTransaction;
```

Comment:
Casting away immutable is bad practice. If we really need mutability we can do `UnitTestGenesisTransaction.serializeFull.deserializeFull!Transaction`.

Reply:
That's a good suggestion. `serializeFull.deserializeFull`